subsys/bluetooth/host/att.c

```diff
@@ -316,6 +384,18 @@ static int bt_att_chan_send(struct bt_att_chan *chan, struct net_buf *buf,
 	return chan_send(chan, buf, cb);
 }
 
+static void bt_att_chan_send_rsp(struct bt_att_chan *chan, struct net_buf *buf,
+				 bt_att_chan_sent_t cb)
+{
+	int err;
+
+	err = bt_att_chan_send(chan, buf, cb);
+	if (err) {
+		/* Responses need to be sent back using the same channel */
+		k_fifo_put(&chan->tx_queue, buf);
```

Comment:
This needs to be `net_buf_puf()`, so that fragment handling is correct. Even if you don't have fragments it ensures that buf->flags, etc is correct.

Reply:
Done.